### ISSUE TITLE
💅 new v2.x with astro file emit false positive unused var and imports

### ISSUE THREAD

**Issue by EKliptus** [biomejs/biome#6519]
### Environment information

```bash
CLI:
  Version:                      2.0.5
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.15

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            false
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnusedImports ; lint/correctness/noUnusedVariables

### Playground link

All astro project with biome v2+ I supposed

### Expected result

All vars, consts and import in frontmatter part of astro file and used in the template part must recognize as used and biome don't need to say it is unused.

Since I migrate from biome v1.9.4 to v2.0.5 I have many lint errors appears on my astro files : all are false positives !

All import and vars declare and unused in the frontmatter part of the astro file but really used in the template part of the same file is emitted as unused.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by siketyan** [comment]
Support for Astro and other HTML-like languages is currently limited, and those false positives are known limitation. See https://biomejs.dev/internals/language-support/#html-super-languages-support.

**Comment by EKliptus** [comment]
> Support for Astro and other HTML-like languages is currently limited, and those false positives are known limitation. See https://biomejs.dev/internals/language-support/#html-super-languages-support.

Ok thank you, sorry, before migrate I don't see these errors but I think when I migrate with `bunx --bun biome migrate --write` it fixed some errors in my config file and now parse astro but before not...

It's ok for me thank you

**Comment by siketyan** [comment]
That's because two rules `noUnusedVariables` and `noUnusedImports` are now turned on by default, since v2.0. We could disable these rules by default by running `biome migrate` though.
